src/options.rs

```diff
@@ -228,9 +230,12 @@ impl Options {
       );
     }
 
-    let client = Client::new(&rpc_url, auth)
-      .with_context(|| format!("failed to connect to Bitcoin Core RPC at {rpc_url}"))?;
+    // let client = Client::new(&rpc_url, auth);
+    //   .with_context(|| format!("failed to connect to Bitcoin Core RPC at {rpc_url}"))?;
+    let t = transport::CustomTransport::new(&rpc_url, auth);
+    let http_client = client::Client::with_transport(t);
```

Comment:
This should be confiigured via the options, since we don't always want to use this transport.